### PR TITLE
tests: Reproduce issue 14685

### PR DESCRIPTION
**Pull request by serathius**
Verified locally that this reproduces #14685
cc @ahrtr @ptabor @geetasg @tjungblu

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -129,6 +129,27 @@ gofail-disable: install-gofail
...
+reproduce-issue14685: ./bin/etcd-v3.5.5-failpoints
+	cp ./bin/etcd-v3.5.5-failpoints ./bin/etcd
+	GO_TEST_FLAGS='-v --run=TestLinearizability/Issue14685 --count 100 --failfast' make test-linearizability
+
+./bin/etcd-v3.5.5-failpoints:
+	rm -rf /tmp/etcd-release-v3.5.5/
+	mkdir -p /tmp/etcd-release-v3.5.5/
+	cd /tmp/etcd-release-v3.5.5/; \
+	  git clone https://github.com/etcd-io/etcd.git .; \
+	  git checkout v3.5.5; \
```
Comment (ahrtr) [inline Makefile:144]: The comment https://github.com/etcd-io/etcd/pull/14912#discussion_r1044422353 applies to this PR as well.

Comment (serathius) [reply Makefile:144]: Marking this PR as draft until we resolve https://github.com/etcd-io/etcd/pull/14912#discussion_r1044422353

Comment (serathius) [reply Makefile]: Removed the script. PTAL.

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @serathius

**Review by ptabor — APPROVED** [review]

**Comment by serathius** [comment]
Interesting we got an inconsistency on GitHub side. Commit for this PR was merged but not the PR.
https://screenshot.googleplex.com/jGVe4zWhEzetuJD

**Pull request closed without merging**